Let ip block style strings be used for override keys

cilphex opened the pull request.

cilphex committed:
Let ip block style strings be used for override keys

README.md: add CIDR block key example
index.js: require netmask and use it to compare override keys to the ratelimit attribute (ip or xff)
package.json: add netmask 1.0.5
tests/index: add tests for ip address and ip block overrides

defunctzombie commented on index.js:
```diff
@@ -67,6 +68,20 @@ function throttle(options) {
+                override.block = block;
+            }
+        }
+        catch(err) {
```
what about just throwing this error like we do above?

cilphex replied:
A key can be a username from `req.username`, which would fail, but should probably be ignored.

defunctzombie replied:
gotcha, thanks for clarifying!

defunctzombie commented on index.js:
```diff
@@ -99,6 +114,16 @@ function throttle(options) {
             burst = options.overrides[attr].burst;
             rate = options.overrides[attr].rate;
         }
+        else if (options.overrides) {
```
seems like this can just into into the `if` above and the `if` above can have two seconds? one for  instant IP dictionary lookup and the other for iterating the keys? If not, then at least a comment here saying this is where the 'cidr blocks' are handled by looking at all the overrides.

cilphex replied:
Refactored

defunctzombie commented on tests/index.js:
```diff
@@ -79,6 +79,80 @@ describe('ratelimit', function() {
+        .end(function(err, res) {
+            done(err);
+        });
+    });
```
what about a test that shows rejecting from an ip outside the block trying to burst or whatnot?

cilphex replied:
Added at bottom.

cilphex committed:
index.js: refactor override code to be less rendundant and have better comments
tests/index.js: add test ensuring that a block rate limit will not apply to an IP that falls outside its range
